src/main/java/org/isf/admission/manager/AdmissionBrowserManager.java

```diff
@@ -404,13 +415,23 @@ else if (dateOut.isBefore(ad.getAdmDate()) || dateOut.equals(ad.getAdmDate())) {
 									DateTimeFormatter.ISO_LOCAL_DATE.format(invalidEnd))));
 				}
 			}
-
+			List<Disease> diseaseIpdOuts1 = diseaseManager.getDiseaseIpdOut().stream().filter(dis-> dis == admission.getDiseaseOut1()).collect(Collectors.toList());
+			List<Disease> diseaseIpdOuts2 = diseaseManager.getDiseaseIpdOut().stream().filter(dis-> dis == admission.getDiseaseOut2()).collect(Collectors.toList());
+			List<Disease> diseaseIpdOuts3 = diseaseManager.getDiseaseIpdOut().stream().filter(dis-> dis == admission.getDiseaseOut3()).collect(Collectors.toList());
 			if (admission.getDiseaseOut1() == null && admission.getDisDate() != null) {
 				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.admission.pleaseselectatleastfirstdiagnosisout.msg")));
 			} else if (admission.getDiseaseOut1() != null && admission.getDisDate() == null) {
 				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.admission.pleaseinsertadischargedate.msg")));
 			}
-
+			if (admission.getDiseaseOut1() != null && diseaseIpdOuts1.size() > 0) {
+				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.admission.diagnosisout1isnotallowed.msg")));
+			}
```

Comment:
Same here, the error should be thrown if the disease is NOT found in the list, hence when `diseaseIpdOutsX.isEmpty()`, correct? and the != null check shouldn't be done before the `stream().filter`?

Reply:
If the error is thrown then all subsequent checks are skipped.   I thought the routine was set to check for everything and only throw the exception with the *list* of errors.

Reply:
Sorry, I meant to say to be added to `errors` list...

Reply:
If I have understand you suggestion is to check first if diseaseIpdOutsX.isEmpty() check now if admission.getDiseaseOutX() is != null if yes added the error in error list. now if the !diseaseIpdOutsX.isEmpty() now done the stream.filter after check if admission.getDiseaseOut3() != null && diseaseIpdOuts.isEmpty() if yes added the error in error list. Is it please?

Reply:
Sorry, let me close this comment and propose a suggestion